### PR TITLE
Workflow backwards compatibility

### PR DESCRIPTION
**Pull request by jacobtoppm**
Adds migrations to the new workflow system, including a data migration to create the existing moderator permissions via a set of workflows instead. To make this easier, serialize the page manager.

Simplifies the workflow message for the page edit view if the workflow only has one step, and adds the 'compare with live' button to all workflow messages when relevant.

### CODE REVIEW

File: wagtail/admin/tests/pages/test_create_page.py [hunk 1]
```diff
@@ -450,13 +450,9 @@ def test_create_simplepage_post_submit(self):
...
 
-        # The latest revision for the page should now be in moderation
-        self.assertTrue(page.get_latest_revision().submitted_for_moderation)
+        # The page should now be in moderation
+        self.assertEqual(page.current_workflow_state.status, page.current_workflow_state.STATUS_IN_PROGRESS)
 
-        # Check that the moderator got an email
-        self.assertEqual(len(mail.outbox), 1)
-        self.assertEqual(mail.outbox[0].to, ['moderator@email.com'])
-        self.assertEqual(mail.outbox[0].subject, 'The page "New page!" has been submitted for moderation')
```
Comment (gasman) [inline wagtail/admin/tests/pages/test_create_page.py]: What's the rationale for removing this? (If the old implementation sent out an email at this point whereas the new one doesn't, isn't that an issue we need to address?)

Comment (jacobtoppm) [reply wagtail/admin/tests/pages/test_create_page.py]: The new one does, but this is tested more thoroughly in `test_workflows`

File: wagtail/core/migrations/0048_add_default_workflows.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    model='groupapprovaltask', app_label='wagtailcore')
+
+    publish_permissions = GroupPagePermission.objects.filter(permission_type='publish')
+
+    for permission in publish_permissions:
+        
+        # find groups with publish permission over this page or its ancestors (and therefore this page by descent)
+        page = permission.page
+        page = Page.objects.get(pk=page.pk)
+        Page.steplen = 4
```
[the remark this answers is not in the document]
Comment (jacobtoppm) [reply wagtail/core/migrations/0048_add_default_workflows.py]: Good point! Will do this

[the remark this answers is not in the document]
Comment (jacobtoppm) [reply wagtail/core/migrations/0048_add_default_workflows.py]: Changed now, thanks!

**Pull request merged**